python/spglib/spglib.py

```diff
@@ -97,6 +103,101 @@ class SpglibError:
 spglib_error = SpglibError()
 
 
+@dataclasses.dataclass(eq=True, frozen=True)
+class SpaceGroupType(Mapping[str, Any]):
+    """Space group type information.
+
+    Dictionary keys are as follows:
+
+    - number : int
+        International space group number
```

Comment:
Can try to move these into the property documentation. For sphinx the following should work:
```
    number: int
    """International space group number"""
```
(docstring has to be after the attribute here)

Reply:
Do you know how to remove unwanted "None" in the sphinx-generated docs?
![image](https://github.com/spglib/spglib/assets/7272505/f590ab7a-084a-40ea-b942-fab24d3b0209)

Reply:
It looks like it is trying to document the `default` value, for which there is none.
`````````markdown
`````{py:class} SpaceGroupType
:canonical: spglib.SpaceGroupType

Bases: {py:obj}`spglib.DictInterface`

```{autodoc2-docstring} spglib.SpaceGroupType
:parser: rst
```

````{py:attribute} number
:canonical: spglib.SpaceGroupType.number
:type: int
:value: >
   None

```{autodoc2-docstring} spglib.SpaceGroupType.number
:parser: rst
```
````
`````````

This seems to be a [bug in `autodoc2`](https://github.com/sphinx-extensions2/sphinx-autodoc2/blob/19587f0f56a17f6f9cc1c2c5614ad8c98d7293ff/src/autodoc2/render/myst_.py#L347-L362). Issue is `myst` and associated team are focusing right now on `mystmd` (btw great project you should follow), so addressing these bugs might be a bit hard.

This could be resolved in #466. In principle that PR is works fine at least as a direct replacement to the current features. It's just the type-annotation that I was looking at addressing, I didn't manage to nail it down.

Reply:
I see... Then, let's leave it for now.